axlearn/common/attention.py

```diff
@@ -1781,8 +1832,6 @@ def _forward_for_mode(
                 "key and value must be both None or both set, "
                 f"key:{type(key)}, value:{type(value)}"
             )
-        if self._mask_fn and (key is not None or value is not None):
-            raise ValueError("key and value are not expected when using an attention mask.")
```

Comment:
Can you remind me why this logic is removed?

Reply:
It breaks [this test](https://github.com/apple/axlearn/blob/9ba319391c9019df2ff56e192320c45c8808c166/axlearn/common/flash_attention/layer_test.py#L498) where `key` and `value` are specified. It doesn't seem to me that this restriction is necessary, as `MaskFn` does expect `key_position` as well. WDYT?

Reply:
Got it. Thanks!